toshling/_endpoints.py

```diff
@@ -1,10 +1,33 @@
+from urllib.parse import urlparse, parse_qs
+
 from .models import argument_types, return_types
 
 
 class Endpoint:
     def __init__(self, client):
         self.client = client
 
+    def iterate(self, **kwargs):
+        ''' Transparently handle pagination.
+
+        Equivalent to calling list(page=0), list(page=1), ...
+        '''
+        items, response = self.list(**kwargs, return_response=True)
+        while True:
+            for item in items:
+                yield item
+
+            if 'next' not in response.links or 'url' not in response.links['next']:
+                break
+
+            url = response.links['next']['url']  # Of form '/tags?per_page=31&page=1'
+            try:
+                page = int(parse_qs(urlparse(url).query)['page'][0])
```

Comment:
Stuff like this query parsing would go away if the generator that did the iteration lived in the `Client` class.